ykbh/src/lib.rs

```diff
@@ -99,14 +99,18 @@ impl SIRInterpreter {
                 unsafe { dest_ptr.add(usize::try_from(*off).unwrap()) }
             }
             IPlace::Indirect { ptr, off, ty: _ty } => {
-                // Get a pointer to the Indirect.
-                let dest_ptr = self.local_ptr(&ptr.local) as *mut usize;
-                // Dereference the pointer by reading its value.
-                let val = unsafe { std::ptr::read::<usize>(dest_ptr) };
-                // Add the offsets of the Indirect.
-                let ptr = val + usize::try_from(ptr.off).unwrap() + usize::try_from(*off).unwrap();
+                // Get a pointer to the Indirect, which itself points to another pointer.
+                let dest_ptr = self.local_ptr(&ptr.local) as *mut *mut u8;
+                let ptr = unsafe {
+                    // Dereference the pointer, by reading its value.
+                    let mut p = std::ptr::read::<*mut u8>(dest_ptr);
+                    // Add the offsets of the Indirect.
```

Comment:
I think all of the offsets are `usize`s, right? In which case you can do `p.add(ptr.off)`. https://doc.rust-lang.org/std/primitive.pointer.html#method.add

Reply:
Unfortunately, the offsets are `i32`s, so `add` doesn't improve things.